crates/polars-core/src/chunked_array/from.rs

```diff
@@ -207,7 +207,10 @@ where
         #[cfg(debug_assertions)]
         {
             if !chunks.is_empty() && dtype.is_primitive() {
-                assert_eq!(chunks[0].data_type(), &dtype.to_physical().to_arrow(true))
+                assert_eq!(
+                    chunks[0].data_type().underlying_physical_type(),
+                    dtype.to_arrow(true)
+                )
```

Comment:
The data type of the chunks can be non-primitive here, e.g. `Date`. This should be converted to physical for this assertion.

Unrelated improvement: `dtype` has been checked to be primitive, so we do not need `to_physical()` on top of it.